tools/NuGetPackager/PackCommand.cs

```diff
@@ -0,0 +1,146 @@
+                {
+                    continue;
+                }
+
+                // TODO: Efficiency
+                var dependencyDescription = context.LibraryManager.GetLibraries().First(l => l.RequestedRanges.Contains(dependency));
+
+                // REVIEW: Can we get this far with unresolved dependencies
+                if (dependencyDescription == null || !dependencyDescription.Resolved)
+                {
+                    continue;
+                }
+
+                if (dependencyDescription.Identity.Type == LibraryType.Project &&
+                    ((ProjectDescription)dependencyDescription).Project.EmbedInteropTypes)
+                {
+                    continue;
+                }
+
+
+                VersionRange dependencyVersion = null;
+
+                if (dependency.VersionRange == null ||
+                    dependency.VersionRange.IsFloating)
+                {
+                    dependencyVersion = new VersionRange(dependencyDescription.Identity.Version);
+                }
+                else
+                {
+                    dependencyVersion = dependency.VersionRange;
+                }
+
+                Reporter.Verbose.WriteLine($"Adding dependency {dependency.Name.Yellow()} {VersionUtility.RenderVersion(dependencyVersion).Yellow()}");
+
+                yield return new Tuple<string, string>(dependency.Name, dependencyVersion.MinVersion.ToString());
+            }
+        }
+
+        private static string GetLockFileVersion(ProjectContext project, string name) =>
+            project
+                .LockFile
+                .PackageLibraries
+                .First(l => l.Name.Equals(name))
+                .Version
+                .ToNormalizedString();
+
+        private async Task Nuget(params string[] args)
+        {
+            var pInfo = new ProcessStartInfo
+            {
+                Arguments = ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(args),
+                FileName = await GetNugetExePath()
+            };
+            Console.WriteLine("command:   ".Bold().Blue() + pInfo.FileName);
+            Console.WriteLine("arguments: ".Bold().Blue() + pInfo.Arguments);
+
+            Process.Start(pInfo).WaitForExit();
+        }
+
+        private async Task<string> GetNugetExePath()
+        {
+            if (Environment.GetEnvironmentVariable("KOREBUILD_NUGET_EXE") != null)
+            {
+                return Environment.GetEnvironmentVariable("KOREBUILD_NUGET_EXE");
+            }
+
+            var nugetPath = Path.Combine(_baseDir, ".build", "nuget.3.5.0-rc1.exe");
+            if (File.Exists(nugetPath))
+            {
+                return nugetPath;
+            }
+            Console.WriteLine("log : Downloading nuget.exe 3.5.0-rc1".Bold().Black());
+            var response = await new HttpClient().GetAsync("https://dist.nuget.org/win-x86-commandline/v3.5.0-rc1/NuGet.exe");
```

Comment:
YAGNI. We need NuGet ≤ 3.5.0-rc1 to get package type support in nuspec. By the time 3.5.0 rtm or 3.6.0 come out, we'll have removed this anyways. See https://github.com/aspnet/DotNetTools/issues/179